vehicle/tesla-vehicle-command/identity.go

```diff
@@ -1,8 +1,16 @@
 package vc
 
 import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+
+	"github.com/evcc-io/evcc/server/db/settings"
```

Comment:
Vom der Modularität und Testbarkeit find ich besser, wenn wir keine direkten Abhängigkeiten auf die Persistenz/Settings haben sondern den Zugriff auf Settings runterreichen. Das sind dann zwar mehr Parameter, dafür ists aber entkoppelt (bspw. über ein Interface).

Reply:
Ich auch. Aber wie?

Reply:
Was der eleganteste Weg ist kann ich dir nicht sagen. Aber im Auth PR hab ich nen einfachen Versuch gemacht und Settings nur in `site` instanziiert und dann runtergereicht.
Im Vehicle Fall müsste man das nach diesem Muster dann natürlich allen Vehicles mitgeben (obwohl die das fast alle nicht brauchen). Du bist tiefer in der Sprache. Vielleicht können Vehicles und andere Devices auch optional ein `WithSettings` Interface implementieren über dass ihnen (sofern implementiert) dann beim Erzeuger eine Instanz gegeben wird über die sie persistieren können.

Reply:
Hab gesehen, dass es auch für Go Dependency Injection Bibliotheken gibt. Bspw. [google/wire](https://github.com/google/wire). Damit könnte man etwas Boilerplate loswerden.